### PR TITLE
fixing code example and adding clarifications to docs

### PR DESCRIPTION
**Pull request by RanVaknin**
This doc PR addresses the following documentation issues:
- https://github.com/aws/aws-sdk-go-v2/issues/2775
- https://github.com/aws/aws-sdk-go-v2/issues/2736
- https://github.com/aws/aws-sdk-go-v2/issues/1173
- https://github.com/aws/aws-sdk-go-v2/issues/2780

### CODE REVIEW

File: content/en/docs/migrating/_index.md [hunk 1]
```diff
@@ -34,9 +34,9 @@ Import Root | Description
 
 ## Configuration Loading
 
-The [session]({{< apiref v1="aws/session" >}}) package and associated functionality are replaced with a simplified
-configuration system provided by the [config]({{< apiref "config" >}}) package. The `config` package is a separate Go
-module, and can be included in your applications dependencies by with `go get`.
+The [session]({{< apiref v1="aws/session" >}}) package from AWS SDK for Go v1 has been replaced by the [config]({{< apiref "config" >}})
+package in AWS SDK for Go v2. You can include this new config package in your application's dependencies with go get.
+Both `session.NewSession` in v1 and `config.LoadDefaultConfig` in v2 use the default credential provider chain, which automatically caches and refreshes credentials before they expire.
```
Comment (lucix-aws) [inline content/en/docs/migrating/_index.md:39]: This blurb doesn't really fit in the context of migrating. I get that you're trying to answer one of the questions in one of the issues this addresses, but comments about the nature of `config` really belong elsewhere.

Comment (RanVaknin) [reply content/en/docs/migrating/_index.md]: Agreed. I relocated this clarification the `info` blob clarifying the behavior of `LoadDefaultConfig` and the cache behavior.

File: docs/docs/code-examples/cloudwatch/index.xml [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
   <channel>
     <title>AWS SDK for Go V2 – Amazon CloudWatch Examples</title>
-    <link>https://aws.github.io/aws-sdk-go-v2/docs/code-examples/cloudwatch/</link>
+    <link>http://localhost:1313/aws-sdk-go-v2/docs/code-examples/cloudwatch/</link>
     <description>Recent content in Amazon CloudWatch Examples on AWS SDK for Go V2</description>
     <generator>Hugo -- gohugo.io</generator>
     <language>en</language>
     
-	  <atom:link href="https://aws.github.io/aws-sdk-go-v2/docs/code-examples/cloudwatch/index.xml" rel="self" type="application/rss+xml" />
+	  <atom:link href="http://localhost:1313/aws-sdk-go-v2/docs/code-examples/cloudwatch/index.xml" rel="self" type="application/rss+xml" />
```
Comment (stanhu) [inline docs/docs/code-examples/cloudwatch/index.xml:9]: Are these URL changes intended?

Comment (lucix-aws) [reply docs/docs/code-examples/cloudwatch/index.xml:9]: No - fixed in #2784

**Pull request merged**